src/Model/TransportModel/tsp1.f90

```diff
@@ -0,0 +1,900 @@
+        write (this%iout, '(4x,a,a)') &
+          'NEWTON-RAPHSON UNDER-RELAXATION based on the bottom ', &
+          'elevation of the model will be applied to the model.'
+      end if
+    end if
+
+    if (found%print_input) then
+      write (this%iout, '(4x,a)') 'STRESS PACKAGE INPUT WILL BE PRINTED '// &
+        'FOR ALL MODEL STRESS PACKAGES'
+    end if
+
+    if (found%print_flows) then
+      write (this%iout, '(4x,a)') 'PACKAGE FLOWS WILL BE PRINTED '// &
+        'FOR ALL MODEL PACKAGES'
+    end if
+
+    if (found%save_flows) then
+      write (this%iout, '(4x,a)') &
+        'FLOWS WILL BE SAVED TO BUDGET FILE SPECIFIED IN OUTPUT CONTROL'
+    end if
+
+    write (this%iout, '(1x,a)') 'END NAMEFILE OPTIONS:'
+  end subroutine log_namfile_options
+
+  !> @brief Source package info and begin to process
+  !<
+  subroutine create_packages(this, indis, gwecommon)
+    ! -- modules
+    use ConstantsModule, only: LINELENGTH, LENPACKAGENAME
+    use CharacterStringModule, only: CharacterStringType
+    use ArrayHandlersModule, only: expandarray
+    use MemoryManagerModule, only: mem_setptr
+    use MemoryHelperModule, only: create_mem_path
+    use SimVariablesModule, only: idm_context
+    use GwfDisModule, only: dis_cr
+    use GwfDisvModule, only: disv_cr
+    use GwfDisuModule, only: disu_cr
+    use TspIcModule, only: ic_cr
+    use TspFmiModule, only: fmi_cr
+    use TspAdvModule, only: adv_cr
+    use TspSsmModule, only: ssm_cr
+    use TspMvtModule, only: mvt_cr
+    use TspOcModule, only: oc_cr
+    use TspObsModule, only: tsp_obs_cr
+    ! -- dummy
+    class(TransportModelType) :: this
+    integer(I4B), intent(inout) :: indis ! DIS enabled flag
+    type(GweInputDataType), intent(in), optional :: gwecommon !< shared data container for use by multiple GWE packages
+    ! -- local
+    type(CharacterStringType), dimension(:), contiguous, &
+      pointer :: pkgtypes => null()
+    type(CharacterStringType), dimension(:), contiguous, &
+      pointer :: pkgnames => null()
+    type(CharacterStringType), dimension(:), contiguous, &
+      pointer :: mempaths => null()
+    integer(I4B), dimension(:), contiguous, &
+      pointer :: inunits => null()
+    character(len=LENMEMPATH) :: model_mempath
+    character(len=LENFTYPE) :: pkgtype
+    character(len=LENPACKAGENAME) :: pkgname
+    character(len=LENMEMPATH) :: mempath
+    integer(I4B), pointer :: inunit
+    integer(I4B) :: n
+    !
+    ! -- Initialize
+    indis = 0
+    !
+    ! -- set input memory paths, input/model and input/model/namfile
+    model_mempath = create_mem_path(component=this%name, context=idm_context)
+    !
+    ! -- set pointers to model path package info
+    call mem_setptr(pkgtypes, 'PKGTYPES', model_mempath)
+    call mem_setptr(pkgnames, 'PKGNAMES', model_mempath)
+    call mem_setptr(mempaths, 'MEMPATHS', model_mempath)
+    call mem_setptr(inunits, 'INUNITS', model_mempath)
+    !
+    do n = 1, size(pkgtypes)
```

Comment:
@langevin-usgs, I assume that what you mean here is that this one routine is to be further split into 2 routines sometime in future?  Or did you mean for me to tackle this now?  Happy to oblige, but wanted to ask if this is something you wanted to chat about in person before I address this?  Presumably, this would also impact GWF, since a similar refactor could be implemented there.

Reply:
I don't think you need to take this on now, but it seems like we might be able to split this at some point.